Improve iam_group exception handling

willthames opened the pull request:
##### SUMMARY
Use AnsibleAWSModule for iam_group and handle BotoCoreErrors
as well as ClientErrors. Use fail_json_aws to improve error messages

Fixes #45593
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
iam_group

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
ansible 2.8.0.dev0 (devel 1463c2e4a8) last updated 2018/09/08 10:08:47 (GMT +1000)
  config file = None
  configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/will/src/ansible/lib/ansible
  executable location = /Users/will/src/ansible/bin/ansible
  python version = 2.7.14 (default, Mar  9 2018, 23:57:12) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION
Candidate for backport to all supported versions of ansible

willthames committed:
Improve iam_group exception handling

Use AnsibleAWSModule for iam_group and handle BotoCoreErrors
as well as ClientErrors. Use fail_json_aws to improve error messages

willthames commented:
Unfortunately there is no test suite for iam_group, and I ran out of time to perform full testing on this change, so if anyone has a test suite to hand, I'd be grateful if they could run it in.

I started a test suite so I can always come back to that later.

s-hertel reviewed (approved):
Would be nice to have tests but this looks all very straightforward.

willthames committed:
Add minimal iam_group test suite

Update some of the read-only IAM permissions (this is not sufficient
to run the test suite but it gets further than it did until it tries
to add a (non-existent) user)

willthames commented:
@s-hertel I added a trivial test suite.

s-hertel committed:
Clean up after tests
